lib/multihash.ex

```diff
@@ -227,11 +230,12 @@ defmodule Multihash do
   end
 
   @doc """
-  Checks if the length of the `digest` is same as the expected `default_length` of the hash function while encoding
+  Checks if the length of the `digest` is the default length for the hash function,or at least greater than the the desired truncated length
   """
-  defp check_digest_length([code: _code, length: default_length], digest) when is_binary(digest) do
+  defp check_digest_length([code: _code, length: default_length], digest, trunc_length) when is_binary(digest) do
     case byte_size(digest) do
       ^default_length -> :ok
+      digest_len when digest_len >= trunc_length -> :ok
```

Comment:
This should also check for case when `trunc_length` is not `:default`. As if it is default then this check shouldn't pass.

```
digest_len when trunc_length !== :default and digest_len >= trunc_length -> :ok
```
```